test/extended/util/framework.go

```diff
@@ -41,7 +41,40 @@ var TestContext e2e.TestContextType
 
 const pvPrefix = "pv-"
 
-//CheckOpenShiftNamespaceImageStreams is a temporary workaround for the intermittent
+// WaitForOpenShiftNamespaceImageStreams waits for the standard set of imagestreams to be imported
+func WaitForOpenShiftNamespaceImageStreams(oc *CLI) error {
+	langs := []string{"ruby", "nodejs", "perl", "php", "python", "wildfly", "mysql", "postgresql", "mongodb", "jenkins"}
+	scan := func() bool {
+		for _, lang := range langs {
+			is, err := oc.REST().ImageStreams("openshift").Get(lang)
+			if err != nil {
+				return false
+			}
+			for tag := range is.Spec.Tags {
+				if _, ok := is.Status.Tags[tag]; !ok {
+					return false
+				}
+			}
+		}
+		return true
+	}
+
+	success := false
+	for i := 0; i < 10; i++ {
```

Comment:
This is why we built https://github.com/openshift/origin/blob/master/vendor/k8s.io/kubernetes/pkg/watch/until.go#L35.  You can write a union condition func to cover all of them.